dotnet/samples/GettingStartedWithProcesses/SharedSteps/ScriptedUserInputStep.cs

```diff
@@ -57,12 +60,22 @@ public override ValueTask ActivateAsync(KernelProcessStepState<UserInputState> s
     [KernelFunction(Functions.GetUserInput)]
     public async ValueTask GetUserInputAsync(KernelProcessStepContext context)
     {
-        var userMessage = _state!.UserInputs[_state.CurrentInputIndex];
+        if (_state == null || _state.CurrentInputIndex >= _state.UserInputs.Count)
+        {
+            // Emit the completion event
+            await context.EmitEventAsync(new() { Id = CommonEvents.UserInputComplete });
+            return;
+        }
+
+        string userMessage = _state.UserInputs[_state.CurrentInputIndex];
```

Comment:
The previous code was checking for the `userMessage` to be equal to `ChatBotEvents.Exit` and then emitting the dotnet equivalent of: ```await context.emit_event(KernelProcessEvent(id=ChatBotEvents.Exit, data=None))```